improver/blending/weighted_blend.py

```diff
@@ -912,7 +915,7 @@ def process(self, cube, weights=None, cycletime=None, attributes_dict=None):
         self._update_blended_metadata(result, attributes_dict)
 
         # Re-mask output
-        if isinstance(cube.data, np.ma.core.MaskedArray):
+        if self.has_mask:
             result.data = np.ma.array(result.data)
```

Comment:
@BelligerG, all tests pass for me if I simply delete these lines altogether, so I suggest we do that instead.